motionplan/tpspace/ptgGroupFrame.go

```diff
@@ -32,52 +35,94 @@ var defaultPTGs = []ptgFactory{
 	NewSideSOverturnPTG,
 }
 
+var defaultDiffPTG ptgFactory = NewDiffDrivePTG
+
 type ptgGroupFrame struct {
 	name               string
 	limits             []referenceframe.Limit
 	geometries         []spatialmath.Geometry
 	ptgs               []PTG
+	solvers            []PTGSolver
 	velocityMMps       float64
+	angVelocityRadps   float64
 	turnRadMillimeters float64
 	logger             golog.Logger
 }
 
-// NewPTGFrameFromTurningRadius will create a new Frame which is also a PTGProvider. It will precompute the default set of
+// NewPTGFrameFromKinematicOptions will create a new Frame which is also a PTGProvider. It will precompute the default set of
 // trajectories out to a given distance, or a default distance if the given distance is <= 0.
-func NewPTGFrameFromTurningRadius(
+func NewPTGFrameFromKinematicOptions(
 	name string,
 	logger golog.Logger,
-	velocityMMps, turnRadMeters, refDist float64,
+	velocityMMps, angVelocityDegps, turnRadMeters, refDist float64,
 	geoms []spatialmath.Geometry,
+	diffDriveOnly bool,
 ) (referenceframe.Frame, error) {
 	if velocityMMps <= 0 {
 		return nil, fmt.Errorf("cannot create ptg frame, movement velocity %f must be >0", velocityMMps)
 	}
-	if turnRadMeters <= 0 {
+	if turnRadMeters < 0 {
 		return nil, fmt.Errorf("cannot create ptg frame, turning radius %f must be >0", turnRadMeters)
 	}
 	if refDist < 0 {
 		return nil, fmt.Errorf("cannot create ptg frame, refDist %f must be >=0", refDist)
 	}
+	if diffDriveOnly && turnRadMeters != 0 {
+		return nil, errors.New("if diffDriveOnly is used, turning radius must be zero")
+	}
 
 	turnRadMillimeters := turnRadMeters * 1000
 
+	angVelocityRadps := rdkutils.DegToRad(angVelocityDegps)
+	if angVelocityRadps == 0 {
```

Comment:
We might think about separating all this logic into its own function

Reply:
Would you settle for a TODO on this?